lib/trie/proof/verify_test.go

```diff
@@ -64,7 +66,7 @@ func Test_Verify(t *testing.T) {
 			errWrapped: ErrKeyNotFoundInProofTrie,
 			errMessage: "key not found in proof trie: " +
 				"0x0101 in proof trie for root hash " +
-				"0xec4bb0acfcf778ae8746d3ac3325fc73c3d9b376eb5f8d638dbf5eb462f5e703",
```

Comment:
Is it better in these cases to no longer test against hard-coded values?

Reply:
math/rand is deterministic. It will always give us the same value, that's why we can use hardcoded values. crypto/rand always gives us different values. So, this hash value changes, and thus we can't use hardcoded values.

In general, it is a good idea to use hardcoded values when when possible.

Reply:
Okay this makes sense to me thank you. I was initially concerned that we could run into issues with it no longer being deterministic, but it seems this only affects test files so we should be okay I think.